### PR TITLE
coqPackages.VST: 2.7.1 -> 2.8

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 127079` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coqPackages.VST</li>
    <li>coqPackages.compcert</li>
  </ul>
</details>

**Comment by CohenCyril** [comment]
Please test `coqPackages_8_12` as well...

**Comment by CohenCyril** [comment]
Also, have you read this before https://github.com/NixOS/nixpkgs/pull/126214 ?
+CC @Zimmi48

File: pkgs/development/coq-modules/compcert/default.nix [hunk 1]
```diff
@@ -15,7 +15,8 @@ let compcert = mkCoqDerivation rec {
   releaseRev = v: "v${v}";
 
   defaultVersion =  with versions; switch coq.version [
-      { case = range "8.8" "8.13"; out = "3.8"; }
+      { case = range "8.8" "8.11"; out = "3.8"; }
+      { case = range "8.2" "8.13"; out = "3.9"; }
```
Comment (Zimmi48) [inline pkgs/development/coq-modules/compcert/default.nix:19]:
```suggestion
      { case = range "8.12" "8.13"; out = "3.9"; }
```

**Comment by Zimmi48** [comment]
Once you fix the minor issue I noted, I will trigger a full `coqPackages` test run at https://github.com/coq-community/coq-nix-toolbox.

**Comment by Zimmi48** [comment]
Tests on the coq-nix-toolbox passed :)
(only the unrelated package coq-elpi failed to build for a reason that appears to be spurious)

**Comment by siraben** [comment]
Good to merge?

**Review by CohenCyril — APPROVED** [review]

**Review by Zimmi48 — APPROVED** [review]

**Pull request merged**